provider/parameter.go

```diff
@@ -466,3 +465,10 @@ func ParameterEnvironmentVariable(name string) string {
 	sum := sha256.Sum256([]byte(name))
 	return "CODER_PARAMETER_" + hex.EncodeToString(sum[:])
 }
+
+func validationError(customMessage, defaultMessage string, a ...any) error {
+	if customMessage != "" {
+		return fmt.Errorf(customMessage)
```

Comment:
Do you anticipate template authors to pass format specifiers into this message?

Reply:
I thought about support for `fmt.Sprintf`, but rather in coderd. Something like: `{min}` and `{max}`. WDYT?

Reply:
I'd add `{value}` as well to `{min}` and `{max}`, but that makes sense to me 👍

Reply:
`{value}` would be useful too! I'm wondering whether the message should be rendered in the provider code, or in `coderd` 🤔

Reply:
To be honest, provider seems the logical place as this is is closest to where it's used.